presto-main/src/main/java/com/facebook/presto/execution/buffer/OutputBuffer.java

```diff
@@ -110,6 +112,26 @@
      */
     void fail();
 
+    /**
+     * Notify buffer that no more pages will be added for the given lifespan.
+     * Any future calls to enqueue a page of that lifespan are ignored.
```

Comment:
Is it true?

Reply:
I update the state condition in `enqueue`. it's true now :)